submissions/webdevtoday/oop_exercise/index.js

```diff
@@ -0,0 +1,99 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor(name, gender, saying, species, legs, friends) {
+      this.name = name;
+      this.gender = gender;
+      this.species = species;
+      this.saying = saying;
+      this.friends = friends;
+      this.legs = legs;
+   }
+   getSaying() {
+      return this.saying;
+   }
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.saying, this.friends.join(', ')];
+   }
+   toString() {  
+      return this.getProperties().join('; ');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, friends) {
+      super(name, gender, saying, 'human', 2, friends);
+      this.hands = 2;
+   }
+
+   getProperties() {
+      const props = super.getProperties();
+      props.splice(3, 0, this.hands);
```

Comment:
This approach relies on a particular order of props returned by the `super.getProperties`.
In reality a peer developer may change super class implementation legally taking no care about child classes (this is one of the points of OOP), so that this fragment may result in unexpected outcome.
Beware of such risks and design your code so that each class doesn't depend on particular implementations in neither parent nor child classes.

Reply:
> This approach relies on a particular order of props returned by the `super.getProperties`.
> In reality a peer developer may change super class implementation legally taking no care about child classes (this is one of the points of OOP), so that this fragment may result in unexpected outcome.
> Beware of such risks and design your code so that each class doesn't depend on particular implementations in neither parent nor child classes.

That's true, I'll think about it